app/scripts/application.js

```diff
@@ -1,41 +1,33 @@
 import Backbone from 'backbone';
 import Marionette from 'backbone.marionette';
-import AnimatedRegion from 'marionette-animated-region';
 import AppConfig from 'scripts/config';
 import LinkOverride from 'scripts/overrides/link';
+import RootLayout from 'scripts/views/layouts/root_layout';
+import routes from 'scripts/helpers/routes';
+import { props } from 'scripts/decorators';
 
-const App = new Marionette.Application();
-
-App.history = () => {
-  Backbone.history.start({
-    pushState: true,
-    root: AppConfig.rootPath
-  });
-};
+class App extends Marionette.Application {
+  initialize() {
+    this.layout = new RootLayout();
+    this.origin = Backbone.history.location.origin;
+  }
 
-App.origin = Backbone.history.location.origin;
+  onBeforeStart() {
+    routes.setRootPath(AppConfig.rootPath);
```

Comment:
I want to move routes helper out of the backbone-base in the near future (as npm-package). So I decided to get rid from AppConfig dependency inside helper and move it to `setRootPath` method before the app has started.